solidity/dashboard/src/sagas/staking.js

```diff
@@ -166,15 +182,92 @@ function* fetchTopUps() {
     }
 
     yield put({ type: "staking/fetch_top_ups_start" })
-    const { delegations, undelegations } = yield select(
+    const { delegations, undelegations, initializationPeriod } = yield select(
       (state) => state.staking
     )
     const operators = [...undelegations, ...delegations].map(
       ({ operatorAddress }) => operatorAddress
     )
     const topUps = yield call(fetchAvailableTopUps, operators)
-    yield put({ type: "staking/fetch_top_ups_success", payload: topUps })
+    yield put({
+      type: "staking/fetch_top_ups_success",
+      payload: topUps.map((_) => ({
+        ..._,
+        readyToBeCommitted: isTopUpReadyToBeCommitted(_, initializationPeriod),
+      })),
+    })
   } catch (error) {
     yield* logError("staking/fetch_top_ups_failure", error)
   }
 }
+
+export function* watchTopUpReadyToBeCommitted() {
+  // Waiting for top-ups data.
+  yield take("staking/fetch_top_ups_success")
+  yield call(notifyTopUpReadyToBeCommitted)
+
+  while (true) {
+    // Every 5 minutes.
+    yield delay(moment.duration(5, "minutes").asMilliseconds())
+    yield call(notifyTopUpReadyToBeCommitted)
+  }
+}
+
+function* notifyTopUpReadyToBeCommitted() {
+  const topUps = yield select((state) => state.staking.topUps)
+  const initializationPeriod = yield select(
+    (state) => state.staking.initializationPeriod
+  )
+
+  const topUpsReadyToCommit = topUps.filter((topUp) =>
+    isTopUpReadyToBeCommitted(topUp, initializationPeriod)
+  )
+
+  if (isEmptyArray(topUpsReadyToCommit)) {
+    return
+  }
+
+  yield put({
+    type: "staking/top_ups_ready_to_be_committed",
+    payload: topUpsReadyToCommit,
+  })
+
+  const { delegations } = yield select((state) => state.staking)
+
+  let isFromLiquidTokens = false
+  for (const { operatorAddress } of topUpsReadyToCommit) {
+    if (!isFromLiquidTokens) {
+      isFromLiquidTokens = delegations.some(
+        (_) =>
+          isSameEthAddress(_.operatorAddress, operatorAddress) && !_.isFromGrant
+      )
+    }
+    const stake = delegations.find(
+      (_) =>
+        isSameEthAddress(_.operatorAddress, operatorAddress) && _.isFromGrant
+    )
+
+    if (stake) {
+      yield put(
+        showMessage({
+          messageType: messageType.TOP_UP_READY_TO_BE_COMMITTED,
+          messageProps: {
+            sticky: true,
+            grantId: stake.grantId,
+          },
+        })
+      )
+    }
+  }
+
+  if (isFromLiquidTokens) {
+    yield put(
+      showMessage({
+        messageType: messageType.TOP_UP_READY_TO_BE_COMMITTED,
+        messageProps: {
+          sticky: true,
+        },
+      })
+    )
+  }
```

Comment:
DIsplay this also `if (topUpReadeToBeCommittedNotificationAlreadyDisplayed === true)`.

Similar as above.